api/checks/update.go

```diff
@@ -172,8 +172,12 @@ func getDiffSummary(aeAPI shared.AppEngineAPI, diffAPI shared.DiffAPI, baseRun,
 	checkState := summaries.CheckState{
 		TestRun: &headRun,
 		Product: shared.ProductSpec{
-			ProductAtRevision: headRun.ProductAtRevision,
-			Labels:            labels,
+			// [browser]@[sha] is plenty specific, and avoids bad version strings.
```

Comment:
This is probably not specific enough in some cases.

Example: when `headRun` is a PR run, we would only pass `browser[channel]@revision` to `GetMasterDiffURL`, which will match to both `pr_base` and `pr_head` runs, and the users would always expect the `pr_head` run.

Reply:
This is deliberately the case. This only needs to identify any _runs_ precisely enough for a given check; i.e. the CheckState's product is specific enough to identify chrome[experimental], and what the means as a check depends on what's been run (e.g. `master` vs an earlier `master`, `pr_base` vs `pr_head`, etc.)

As for below, let's rejig the API to take a run to diff against master, and then we can be less ambiguous :)